CMSgov/easi-app
# Easi 1991/cedar url info

# EASI-1991

## Changes and Description

- Allow querying URL info for systems from CEDAR. This can be queried with a top-level `urls` query or as part of the existing `cedarSystemDetails` query.
- Include threat info in the `CedarSystemDetails` GQL type, so one `cedarSystemDetails` query can fetch all relevant data on a system.
- Make the count* fields on the `CedarAuthorityToOperate` GQL type non-nullable, defaulting to returning a value of 0.

## How to test this change

Test queries to run in the GraphQL playground (when connected to CEDAR Impl):

Should return a single URL entry:
```
query {
  urls(cedarSystemId:"408-4109-0") {
    id
    address
    isBehindWebApplicationFirewall
    isAPIEndpoint
    isVersionCodeRepository
    urlHostingEnv
  }
}
```

Should return multiple URL entries:
```
query {
  urls(cedarSystemId:"408-995-0") {
    id
    address
    isBehindWebApplicationFirewall
    isAPIEndpoint
    isVersionCodeRepository
    urlHostingEnv
  }
}
```

Should return info on multiple threats and multiple URLs, as part of the larger `cedarSystemDetails` query:
```
query {
  cedarSystemDetails(cedarSystemId: "408-1157-0") {
    threats {
      id
      parentId
      type
      weaknessRiskLevel
      controlFamily
      daysOpen
    }
    urls {
      id
      address
      isBehindWebApplicationFirewall
      isAPIEndpoint
      isVersionCodeRepository
      urlHostingEnv
    }
  }
}
```
